src/services/graphqlWithProps.tsx

```diff
@@ -0,0 +1,12 @@
+import graphql from 'react-apollo/graphql';
+import React from 'react';
+
+/**
+ * a HOC that allows for the building of dynamic queries.
+ * Pass it a function that returns a query and it will invoke it with the wrapped component's props,
+ * creating a graphql component with the returned query
+ */
+export default (query) => (WrappedComponent) => (props) => { // todo better typing
```

Comment:
I'd suggest to do the better typing todo inside this before merging to develop. If you extend it using partials you can also make the queries more typesafe